fix #47 Add Pool#disposeLater for async disposal

simonbasle opened the pull request.

ttddyy commented on src/main/java/reactor/pool/Pool.java:
```diff
@@ -112,4 +113,33 @@
+     * <ul>
+     *     <li>
+     *         notifying every acquire still pending that the pool has been shut down,
+     *         via a {@link RuntimeException}This method <strong>blocks</strong> until all
```
nit: comment seems mixed up.

simonbasle replied:
fixed

mp911de commented on src/main/java/reactor/pool/Pool.java:
```diff
@@ -112,4 +113,33 @@
+     * Otherwise, it performs the same steps as in the blocking counterpart, {@link #dispose()}.
+     * <p>
+     * If the pool has been already shut down, returns {@link Mono#empty()}. Completion of
+     * the {@link Mono} indicate completion of the shutdown process.
```
Nit: `indicates`

simonbasle replied:
fixed

simonbasle commented:
@mp911de in the first iteration the behavior of `dispose()` changed due to switching to `.disposeLater().block()` (whereas master is just doing `.subscribe()` on each release `Mono`). I changed that to `.diposeLater().subscribe()` (and updated the javadoc accordingly). Can you please re-review? 🙇

mp911de commented on src/main/java/reactor/pool/Pool.java:
```diff
@@ -112,4 +113,39 @@
+     * <ul>
+     *     <li>
+     *         notifying every acquire still pending that the pool has been shut down,
+     *         via a {@link RuntimeException}
```
Probably a `IllegalStateException` (or a dedicated `PoolShutdownException`) would improve debugging experience. A raw `RuntimeException` doesn't feel appropriate.

simonbasle replied:
yes, I'll introduce pool-specific exceptions in another PR I think.
